colcon_cmake/task/cmake/__init__.py

```diff
@@ -208,3 +211,64 @@ def get_visual_studio_version():
     :rtype: str
     """
     return os.environ.get('VisualStudioVersion', None)
+
+
+"""
+Global variable for the cached CMake version number.
+
+When valid, this will be a pkg_resources.extern.packaging.version.Version.
+It may also have a boolean value False when the CMake version could not be
+determined to avoid trying to determine it again.
+"""
+_cached_cmake_version = None
+
+
+def get_cmake_version():
+    """
+    Get the CMake version.
+
+    The function caches the result on the first invocation and reused that on
+    subsequent invocations.
+    The result is None when a version number could not be determined.
+
+    :returns: The version as reported by `CMAKE_EXECUTABLE --version`, or None
+    :rtype pkg_resources.extern.packaging.version.Version
+    """
+    global _cached_cmake_version
+    if _cached_cmake_version is None:
+        cmake_version = _parse_cmake_version()
+        if cmake_version is not None:
+            _cached_cmake_version = cmake_version
+        else:
+            # flag to avoid retrying on subsequent calls
+            _cached_cmake_version = False
+
+    if _cached_cmake_version is False:
+        # in case the CMake version could not be determined
+        return None
+
+    return _cached_cmake_version
+
+
+def _parse_cmake_version():
+    """
+    Parse the CMake version printed by `CMAKE_EXECUTABLE --version`.
+
+    :returns: The version parsed by pkg_resources.parse_version, or None
+    :rtype pkg_resources.extern.packaging.version.Version
+    """
+    try:
+        output = subprocess.check_output(
+            [CMAKE_EXECUTABLE, '--version'], stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        print('Failed to determine CMake version: ' + e.output.decode(),
+              file=sys.stderr)
+    else:
+        lines = output.decode().splitlines()
+        if lines:
+            # Extract just the version part of the string.
+            ver_re_str = r'^(?:.*)(\d+\.\d+\..*)'
+            ver_match = re.match(ver_re_str, lines[0], re.I)
```

Comment:
Why use `re.I` here? The pattern doesn't seem to contain any alphabetical characters.

Reply:
@KazNX Are planning to update the patch accordingly?